src/plugins/intel_cpu/src/emitters/x64/jit_snippets_emitters.cpp

```diff
@@ -856,48 +880,147 @@ void BrgemmEmitter::initBrgemm(brgemmCtx& ctx, std::unique_ptr<brgemm_kernel_t>&
+}
+
+void BrgemmEmitter::emit_N_blocking_loops(size_t k_kernel_id,
+                                          const Xbyak::Reg64& input_0, const Xbyak::Reg64& input_1,
+                                          const Xbyak::Reg64& input_2, const Xbyak::Reg64& output_0,
+                                          const Xbyak::Reg64& work_amount_N) const {
+    auto emit_and_shift_pointers = [&](size_t kernel_idx) {
+        const auto& brgemmCtx = m_brgCtxs[kernel_idx];
+        emit_brgemm_kernel_call(m_brgKernels[kernel_idx].get(), brgemmCtx, input_0, input_1, input_2, output_0);
+        h->add(output_0, brgemmCtx.N * io_data_size.back());
+        h->add(input_1, brgemmCtx.N * io_data_size[1]);
+        if (m_with_scratch && m_with_comp)
+            h->add(input_2, brgemmCtx.N * io_data_size[2]);
+    };
+    // Blocked N loop
+    size_t kernel_idx = getBrgIdx(k_kernel_id, 0);
+    if (m_brgKernels[kernel_idx]) {
+        const auto& brgemmCtx = m_brgCtxs[kernel_idx];
+        Label N_loop_begin;
+        if (m_N_blk_loop) {
+            h->mov(work_amount_N, m_N);
+            h->L(N_loop_begin);
+        }
+
+        emit_and_shift_pointers(kernel_idx);
+
+        if (m_N_blk_loop) {
+            h->sub(work_amount_N, brgemmCtx.N);
+            h->cmp(work_amount_N, brgemmCtx.N);
+            h->jge(N_loop_begin);
+        }
+    }
+    // N loop tail
+    kernel_idx = getBrgIdx(k_kernel_id, 1);
+    if (m_brgKernels[kernel_idx])
+        emit_brgemm_kernel_call(m_brgKernels[kernel_idx].get(), m_brgCtxs[kernel_idx], input_0, input_1, input_2, output_0);
+
+    h->sub(input_1, (m_N - m_N_tail) * io_data_size[1]);
+    h->sub(output_0, (m_N - m_N_tail) * io_data_size.back());
+    if (m_with_scratch && m_with_comp)
+        h->sub(input_2, (m_N - m_N_tail) * io_data_size[2]);
+}
+
 void BrgemmEmitter::emit_impl(const std::vector<size_t>& in,
                               const std::vector<size_t>& out) const {
     if (host_isa_ == cpu::x64::avx512_core) {
         Xbyak::Reg64 input_0(static_cast<int>(in[0]));
         Xbyak::Reg64 input_1(static_cast<int>(in[1]));
         Xbyak::Reg64 input_2(static_cast<int>(0));  // scratch. Default reg index is 0 if there isn't scratch
+        Xbyak::Reg64 output_0(static_cast<int>(out[0]));
+        Xbyak::Reg64 work_amount_N(static_cast<int>(0));
+        Xbyak::Reg64 work_amount_K(static_cast<int>(0));
+
+        std::set<int> unique_ids{input_0.getIdx(), input_1.getIdx(), output_0.getIdx()};
+        size_t unique_ids_count = 3;
+
+        if (m_N_blk_loop || m_K_blk_loop) {
+            if (aux_gpr_idxs.size() < static_cast<size_t>(m_N_blk_loop) + static_cast<size_t>(m_K_blk_loop))
+                IE_THROW() << "BRGEMM Emitter requires extra gpr which was not allocated";
+            if (m_N_blk_loop) {
+                work_amount_N = Xbyak::Reg64(static_cast<int>(aux_gpr_idxs[0]));
+                unique_ids.insert(work_amount_N.getIdx());
+                unique_ids_count++;
```

Comment:
Minor: We can create lambda function here since we use it 3 times here

Reply:
Done, thanks